### PR TITLE
Add override provisioner's option for HCL2

### PR DESCRIPTION
**Pull request by sylviamoss**
This adds the code to make the provisioner's `override` option work with HCL2, and also the HCL2 example to the docs 👍

### CODE REVIEW

File: website/pages/partials/provisioners/common-config.mdx [hunk 1]
```diff
@@ -10,17 +10,33 @@ Parameters common to all provisioners:
...
 
+  In JSON:
   ```json
   {
     "type": "shell",
     "script": "script.sh",
     "override": {
-      "vmware-iso": {
-        "execute_command": "echo 'password' | sudo -S bash {{.Path}}"
+      "example_one": {
```
Comment (SwampDragons) [inline website/pages/partials/provisioners/common-config.mdx:19]: While we're updating this, it kind of makes sense to swap this out for a very small but fully functional build:

```json
{
	"builders": [
		{
			"type": "null",
			"name": "example1",
			"communicator": "none"
		},
		{
			"type": "null",
			"name": "example2",
			"communicator": "none"
		}
	],
	"provisioners": [
		{
			"type": "shell-local",
			"inline": ["echo not overridden"],
			"override": {
				"example2": {
					"inline": ["echo yes overridden"]
				}
			}
		}
	]
}
```

```hcl
source "null" "example1" {
    communicator = "none"
}

source "null" "example2" {
    communicator = "none"
}

build {
    sources = ["source.null.example1", "source.null.example2"]
    provisioner "shell-local" {
        inline = ["echo not overridden"]
        override = {
            example1 = {
                inline = ["echo yes overridden"]
            }
        }
    }
}
```

What do you think?

Comment (SwampDragons) [reply website/pages/partials/provisioners/common-config.mdx:19]: And while I'm on it maybe these would be good files to add to build_test.go which doesn't seem to have any override tests?

Comment (sylviamoss) [reply website/pages/partials/provisioners/common-config.mdx:19]: Both are great!! I liked better your examples, very simple!

**Comment by SwampDragons** [comment]
Awesome 🎉

**Pull request merged**